Funcky.Test/Extensions/EnumerableExtensions/WithFirstTest.cs

```diff
@@ -0,0 +1,43 @@
+using System.Linq;
+using Funcky.Extensions;
+using Xunit;
+
+namespace Funcky.Test.Extensions.EnumerableExtensions
+{
+    public class WithFirstTest
+    {
+        [Fact]
+        public void AnEmptySequenceWithFirstReturnsAnEmptySequence()
+        {
+            var emptySequence = Enumerable.Empty<string>();
+
+            Assert.Empty(emptySequence.WithFirst());
+        }
+
+        [Fact]
+        public void ASequenceWithOneElementWithFirstHasOneElementWhichIsMarkedFirst()
+        {
+            const string expectedValue = "Hello world!";
+            var oneElementSequence = Enumerable.Repeat(expectedValue, 1);
```

Comment:
You can use `Sequence.Return` here (applies to all three test classes)

Reply:
Adressed in 95fcce3a